### PR TITLE
Room details

### LINKED ISSUE #7: [2pt] Create the Room Detail page
- The detail page should have
   - An image of the room
   - Information about the room
   - A reservation button
- Add unit testing.

Image

### PR DESCRIPTION
**Pull request by Zuhaib042**
**Summary**
- In this milestone, I have connected the front-end to API using redux toolkit's createSlice and createAsyncThunk features.
- I have created two slices, one to fetch rooms from API's index method. And one to fetch a single room from API's show method.

### CODE REVIEW

File: src/components/rooms/RoomDetails.js [hunk 1]
```diff
@@ -1,5 +1,61 @@
...
+              <span>{room.city}</span>
+            </li>
+          </ul>
+          <p className="text-sm text-zinc-500 pl-2">{room.description}</p>
+          <button
+            type="button"
+            className="bg-prime py-2 px-4 item self-end text-white rounded-full mt-8"
+          >
+            Reserve
+          </button>
```
Comment (arabiu033) [inline src/components/rooms/RoomDetails.js]: H! @Zuhaib042 Great work so far : +1: Can you make the button here function by navigating a user to the reservation page when clicked?

This is the reservation path - "/reserve/:room_id"

**Review by arabiu033 — CHANGES_REQUESTED** [review]
Hi @Zuhaib042 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦  🏃  🏃‍♀️ 

## Highlights 👍  
- Pages are functioning as expected
- Document work professionally
- Nice and simple design
- Linters are passing

File: src/components/rooms/RoomDetails.js (hunk 1 above)
Comment (Zuhaib042) [reply src/components/rooms/RoomDetails.js]: @arabiu033 done. kindly consider rechecking it

**Review by arabiu033 — APPROVED** [review]
Hi @Zuhaib042  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

**Pull request merged**